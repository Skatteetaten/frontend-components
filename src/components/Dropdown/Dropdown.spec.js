import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { Dropdown } from '.';
import { KeytipData } from '@fluentui/react';

function oppsettShallow(props) {
  return shallow(<Dropdown {...props} />);
}

function oppsettFullDOM(props) {
  return mount(<Dropdown {...props} />);
}

const options = [
  { key: null, text: '' },
  { key: 'Header', text: 'Actions', itemType: Dropdown.ItemType.Header },
  { key: 'A', text: 'Option a' },
  { key: 'B', text: 'Option b' },
  { key: 'C', text: 'Option c' },
  { key: 'D', text: 'Option d' },
  { key: 'E', text: 'Option e' },
  { key: 'divider_2', text: '-', itemType: Dropdown.ItemType.Divider },
  { key: 'Header2', text: 'People', itemType: Dropdown.ItemType.Header },
  { key: 'F', text: 'Option f' },
  { key: 'G', text: 'Option g' },
  { key: 'H', text: 'Option h' },
  { key: 'I', text: 'Option i' },
  { key: 'J', text: 'Option j' },
];

describe('Dropdown komponent', () => {
  it('matcher snapshot', () => {
    const wrapper = oppsettShallow({
      placeholder: 'Select an option',
      options: options,
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('rendrer feilmelding under Dropdown ', () => {
    const wrapper = oppsettShallow({
      errorMessage: 'Feilmelding',
    });

    expect(wrapper.html()).toContain('Feilmelding');
  });

  it('rendrer nedtrekksliste når Dropdown klikkes ', () => {
    const wrapper = oppsettFullDOM({
      options: options,
    });

    const input = wrapper.find('.ms-Dropdown');
    input.simulate('click');
    expect(wrapper.exists('.ms-Dropdown-optionText')).toEqual(true);
    expect(wrapper.exists('.ms-Dropdown-header')).toEqual(true);
    expect(wrapper.exists('.ms-Dropdown-divider')).toEqual(true);
  });
  it('rendrer input med readonly når komponenten har props readOnly ', () => {
    const wrapper = oppsettFullDOM({
      readOnly: true,
      options: options,
      defaultSelectedKey: 'J',
    });
    expect(wrapper.find('input').props().readOnly).toEqual(true);
  });
});
