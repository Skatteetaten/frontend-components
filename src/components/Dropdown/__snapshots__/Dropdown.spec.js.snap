// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown komponent matcher snapshot 1`] = `
<Fragment>
  <div
    className="labelArea-41"
  >
    <span
      className="label-42"
    />
  </div>
  <StyledWithResponsiveMode
    calloutProps={
      Object {
        "className": "SkeDropdownValg-48",
      }
    }
    className="SkeDropdown-47"
    onRenderCaretDown={[Function]}
    options={
      Array [
        Object {
          "key": null,
          "text": "",
        },
        Object {
          "itemType": 2,
          "key": "Header",
          "text": "Actions",
        },
        Object {
          "key": "A",
          "text": "Option a",
        },
        Object {
          "key": "B",
          "text": "Option b",
        },
        Object {
          "key": "C",
          "text": "Option c",
        },
        Object {
          "key": "D",
          "text": "Option d",
        },
        Object {
          "key": "E",
          "text": "Option e",
        },
        Object {
          "itemType": 1,
          "key": "divider_2",
          "text": "-",
        },
        Object {
          "itemType": 2,
          "key": "Header2",
          "text": "People",
        },
        Object {
          "key": "F",
          "text": "Option f",
        },
        Object {
          "key": "G",
          "text": "Option g",
        },
        Object {
          "key": "H",
          "text": "Option h",
        },
        Object {
          "key": "I",
          "text": "Option i",
        },
        Object {
          "key": "J",
          "text": "Option j",
        },
      ]
    }
    placeholder="Select an option"
  />
</Fragment>
`;
